src/vm/lib/tests_helper.c

```diff
@@ -0,0 +1,50 @@
+#include "tests_helper.h"
+
+#include <string.h>
+#include <stdlib.h>
+
+
+char* string_slice(const char input_string[], int start, int end) {
+    int i, j;
+    char *slice_string;
+
+    // Verifies if the start and end positions are valid
+    if(start >= end || end > strlen(input_string)) {
+        return NULL;
+    }
+
+    // Allocate necessary memory for the new slice string
+    slice_string = (char *) malloc(sizeof(char) * (end - start + 1));
+
+    // Copy values from original string to new one
+    for(i = start, j = 0; i < end; i++, j++) {
+        slice_string[j] = input_string[i];
+    }
+
+    // String terminator
+    slice_string[j] = '\0';
+
+    return slice_string;
+}
+
+
+char** arg_str_to_array_of_str(const char* input_string, const char delimiter) {
```

Comment:
The name of this function is too big and it is already implemented in the `vm_strsplit`. It could be implemented as a side function to `vm_strsplit` and I would call this function `vm_strsplit` and the other already implemented `vm_strspliti` to denote the `i` in the end.